src/safeds/data/tabular/containers/_tagged_table.py

```diff
@@ -185,3 +187,631 @@ def _copy(self) -> TaggedTable:
+        """
+        return TaggedTable._from_table(
+            super().add_columns(columns),
+            target_name=self.target.name,
+            feature_names=self.features.column_names,
+        )
+
+    def add_row(self, row: Row) -> TaggedTable:
+        """
+        Add a row to the table.
+
+        This table is not modified.
+
+        Parameters
+        ----------
+        row : Row
+            The row to be added.
+
+        Returns
+        -------
+        table : TaggedTable
+            A new table with the added row at the end.
+
+        Raises
+        ------
+        SchemaMismatchError
+            If the schema of the row does not match the table schema.
+        """
+        return TaggedTable._from_table(super().add_row(row), target_name=self.target.name)
+
+    def add_rows(self, rows: list[Row] | Table) -> TaggedTable:
+        """
+        Add multiple rows to the table.
+
+        This table is not modified.
+
+        Parameters
+        ----------
+        rows : list[Row] or Table
+            The rows to be added.
+
+        Returns
+        -------
+        result : TaggedTable
+            A new table which combines the original table and the given rows.
+
+        Raises
+        ------
+        SchemaMismatchError
+            If the schema of on of the row does not match the table schema.
+        """
+        return TaggedTable._from_table(super().add_rows(rows), target_name=self.target.name)
+
+    def filter_rows(self, query: Callable[[Row], bool]) -> TaggedTable:
+        """
+        Return a table containing only rows that match the given Callable (e.g. lambda function).
+
+        This table is not modified.
+
+        Parameters
+        ----------
+        query : lambda function
+            A Callable that is applied to all rows.
+
+        Returns
+        -------
+        table : TaggedTable
+            A table containing only the rows to match the query.
+        """
+        return TaggedTable._from_table(
+            super().filter_rows(query),
+            target_name=self.target.name,
+            feature_names=self.features.column_names,
+        )
+
+    def keep_only_columns(self, column_names: list[str]) -> TaggedTable:
+        """
+        Return a table with only the given column(s).
+
+        This table is not modified.
+
+        Parameters
+        ----------
+        column_names : list[str]
+            A list containing only the columns to be kept.
+
+        Returns
+        -------
+        table : TaggedTable
+            A table containing only the given column(s).
+
+        Raises
+        ------
+        UnknownColumnNameError
+            If any of the given columns does not exist.
+        IllegalSchemaModificationError
+            If none of the given columns is the target column or any of the feature columns.
+        """
+        if self.target.name not in column_names:
+            raise IllegalSchemaModificationError("Must keep the target column.")
```

Comment:
Consider raising a ColumnIsTargetError here instead

Reply:
I think I had decided against this because the `ColumnIsTargetError` explicitely talks about "removing" a column.
As we're not actively removing columns (that's rather a side effect of us *selecting* and keeping a subset of the columns), I thought the error message format of the `ColumnIsTargetError` doesn't fit 100% and might be confusing.
With the general `IllegalSchemaModificationError`, I have more liberties concerning the exact wording.

I'll let you decide.

Reply:
I see, makes sense to me!